### DISCUSSION TITLE
Impossible to update lockfile in Yarn 3

### DISCUSSION THREAD

**Discussion by 7rulnik** [renovatebot/renovate#10021] [Request Help] [answered]
**What Renovate type, platform and version are you using?**

**Renovate type**

- [ ] Hosted app
- [X] Self hosted

**Renovate platform**

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server (self-hosted)
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server (self-hosted)
- [ ] Gitea (gitea.com, self-hosted)
- [X] GitHub (github.com, GitHub Enterprise)
- [ ] GitLab (gitlab.com, self-hosted)

**Renovate version (if self hosting)**

I'm self hosting Renovate with version: 24.49.9

**Describe the bug**

When using yarn@3 (currently in RC) renovate fails on `yarn install` because it passes `CI=true` environment variable. I debugged it locally and found that problem with `CI=true` flag. Yarn throws with `The lockfile would have been modified by this install, which is explicitly forbidden`. It was introduced in https://github.com/yarnpkg/berry/blob/master/CHANGELOG.md#300-rc1

```
The enableImmutableInstalls will now default to true on CI (we still recommend to explicitly use --immutable on the CLI).
You can re-allow mutations by adding YARN_ENABLE_IMMUTABLE_INSTALLS=false in your environment variables.
```

So seems that it could be workarounded in userland, but I think the right choice is to remove CI env here https://github.com/renovatebot/renovate/blob/main/lib/manager/npm/post-update/yarn.ts#L77

**Relevant debug logs**

There no relevant logs, PR looks like this:

![image](https://user-images.githubusercontent.com/5969049/118477158-0fac1a00-b717-11eb-8d2a-36554c67c495.png)

<details><summary>Click me to see logs</summary>

```
Copy/paste any log here, between the starting and ending backticks
```

</details>

**Have you created a minimal reproduction repository?**

I don't have time right now to create repro, but can do it a bit later.

**Comment by viceice** [comment, marked as answer]
This is fixed in #9515 and [`v24.114.0`](https://github.com/renovatebot/renovate/releases/tag/24.114.0), so you need to update your very outdated renovate.

https://github.com/renovatebot/renovate/blob/9ca604b3453ca55795bd10d3fcad4475c3c8c7cb/lib/manager/npm/post-update/yarn.ts#L97-L97

**Reply by 7rulnik** [reply]
Oh, sorry. Yeah it should help
